File: Sem4/MPP/cs/lab3/ConsoleApp1/Repository/DbAngajatRepo.cs
using System.Data;
using ConsoleApp1.Repository;
using Domain;
using log4net;

namespace Repository;

public class DbAngajatRepo:DbUtils<long,Angajat>,AngajatRepo
{
    private static readonly ILog log = LogManager.GetLogger("AngajatDbRepo");
    
    public DbAngajatRepo(IDictionary<string,string> props):base(props){}

    public void Add(Angajat entity)
    {
        log.Info($"Adding angajat: {entity.Id} ");
        int result = ExecuteNonQuery(
            "insert into angajati (username, password,email) values (@username, @password,@email) ",
            new Dictionary<string, object>
            {
                { "@username", entity.Username },
                { "@password", entity.Password },
                { "@email", entity.Mail }
            });
        if (result == 0)
        {
            log.Error($"Angajat was not added: {entity}");
            throw new RepoException("Angajat was not added");
        }

        log.Info($"Angajat was added");
    }

    public Angajat findByUsernamePassword(string username, string password)
    {
        log.Info($"Finding angajat :{username} {password}");
        return SelectFirst("select * from angajati where username=@username and password=@password",
            new Dictionary<string, object>
            {
                { "@username", username },
                { "@password", password },
            });
    }
    
    public List<Angajat> GetAll()
    {
        log.Info($"Selecting all angajati");
        return Select("select * from angajati").ToList();
    }

    public Angajat GetOne(long id)
    {
        log.Info($"Finding Angajat with id {id}");
        return SelectFirst("select * from angajati where id=@id", new Dictionary<string, object>
        {
            { "@id", id },
        });
    }

    public void Delete(Angajat entity)
    {
        log.Info($"Deleting Angajat with id {entity.Id}");
        var r=ExecuteNonQuery("delete from angajati where id=@id", new Dictionary<string, object>
        {
            { "@id", entity.Id },
        });
        if (r > 0)
            log.Info($"Deleted Angajat with id {entity.Id}");
    }

    public void Update(Angajat e)
    {
        log.Info($"Deleting Angajat with id {e.Id}");
        var r=ExecuteNonQuery("update angajati set email=@email where id=@id", new Dictionary<string, object>
        {
            { "@id", e.Id },
            {"@email", e.Mail },
        });
        if (r > 0)
            log.Info($"Updated Angajat with id {e.Id}");
    }
    

    protected override Angajat DecodeReader(IDataReader reader)
    {
        log.Info($"Deconding Angajat");
        var id = Convert.ToInt32(reader["id"]);
        var username = reader["username"] as string;
        var password = reader["password"] as string;
        var email = reader["mail"] as string;            
        var angajat = new Angajat(username, password, email);
        angajat.Id = id;
        return angajat;
    }
}

File: Sem4/MPP/java/lab11RESTjava/Repository/src/main/java/ubb/scs/mpp/Repository/DbAngajatRepo.java
package ubb.scs.mpp.Repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.springframework.stereotype.Component;
import ubb.scs.mpp.model.Angajat;


import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;
@Component
public class DbAngajatRepo extends DbRepoUtils<Long, Angajat> implements AngajatRepository{
    private static final Logger logger= LogManager.getLogger(DbAngajatRepo.class);

    public DbAngajatRepo(Properties prop) {
        super(prop);
        logger.info("Initializing DbAngajatRepo with properties: {}", prop);
    }

    @Override
    public Angajat decodeResult(ResultSet resultSet) throws SQLException {
        Long id = resultSet.getLong("id");
        String username = resultSet.getString("username");
        String password = resultSet.getString("password");
        String email = resultSet.getString("mail");
        Angajat angajat=new Angajat(username,password,email);
        angajat.setId(id);
        return angajat;
    }

    @Override
    public Angajat findByUsernamePassword(String username, String password) throws RepoException {
        return selectFirst("select * from angajati where username = ? and password = ?", username,password);

    }

    public Angajat findByUsername(String username) throws RepoException {
        return selectFirst("select * from angajati where username = ?", username);
    }

    @Override
    public Angajat findOne(Long aLong) throws RepoException {
        return selectFirst("select * from angajati where id=?",aLong);
    }

    @Override
    public Iterable<Angajat> findAll() throws RepoException {
        return select("select * from angajati");
    }

    @Override
    public Angajat save(Angajat entity) throws RepoException {
        logger.trace("Saving angajat: {}", entity);
        executeNonQuery("insert into angajati (username,password,mail) values (?,?,?)",entity.getUsername(),entity.getPassword(),entity.getMail());
        logger.info("Saved angajat: {}", entity);
        logger.traceExit();
        return entity;
    }

    @Override
    public Angajat delete(Long aLong) throws RepoException {
        logger.trace("Deleting angajat: {}", findOne(aLong));
        executeNonQuery("delete from angajati where id=?",aLong);
        logger.info("Deleted angajat: {}", findOne(aLong));
        logger.traceExit();
        return findOne(aLong);
    }

    @Override
    public Angajat update(Angajat entity) throws RepoException {
        logger.trace("Updating angajat: {}", entity);
        executeNonQuery("update angajati set mail=? where id=? ",entity.getMail(),entity.getId());
        logger.info("Updated angajat: {}", entity);
        logger.traceExit();
        return entity;
    }
}
